automerge/README.md

```diff
@@ -0,0 +1,135 @@
+
+## Tekton Dashboard Webhooks Extension
+
+See the [Getting Started](https://github.com/tektoncd/experimental/blob/master/webhooks-extension/docs/GettingStarted.md) guide for setup guidelines. Our example uses GitHub Enterprise (GHE) and expects webhooks to be delivered to a cluster that is routeable to from GHE.
+
+### Secrets and Service Accounts
+
+Set up your Service Account and Secret as per the guide above. In this case you should have,
+
+- A ServiceAccount configured for use by Tekton.
+- A Tekton-compatible Secret patched onto that that ServiceAccount containing your GitHub token.
+
+This secret is used in two related ways. We check the source repository out using a Tekton Git PipelineResource. This sets up `~/.gitconfig` with the credentials needed for `git push` to work. It gets these credentials from the `accessToken` field of the relevant secret patched onto the ServiceAccount running the Tekton Task. We then extract the same field and export it into the `GITHUB_TOKEN` environment variable to make `hub merge` work. Instructions for creating this secret are in the Getting Started document linked above. You should have resources of the form, 
+
+```yaml
+---
+apiVersion: v1
+data:
+  password: [base64-encoded token]
+  username: [base64-encoded email address]
+kind: Secret
+metadata:
+  annotations:
+    tekton.dev/git-0: https://github.ibm.com  # For example
+  labels:
+    serviceAccount: test-sa                   # As configured in the Webhooks Extension
+  name: github-repo-access-secret
+
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: test-sa
+secrets:
+- name: github-repo-access-secret
+```
+
+### Edit templates
+
+Next edit the `webhooks/templates/*` files.
+
+- In automerge-task.yaml,
+  - replace `YOUR_DOCKER_HUB_ID` with your DockerHub id.
+  - replace `YOUR_GHE` with your GitHub Enterprise domain.
+- In automerge-tt.yaml, replace YOUR_TEKTON_SERVICE_ACCOUNT with the name of your ServiceAccount used by Tekton.
+
+### Apply configuration
+
+Apply your Tekton config as usual: 
+
+```sh
+kubectl apply -f webhooks/resources
+kubectl apply -f webhooks/templates
+```
+
+### Set up webhook, and test
+
+Using the Tekton Dashboard webhooks extension, associate the `automerge-pipeline` with your GitOps repository. Now when a PR is raised against that repo you should see three PipelineRuns created for `automerge-pipeline`. 
+
+- The first is triggered when the branch for the PullRequest is created. This runs the `echo "do nothing"` section in `automerge-task`.
+- The second run executes the bulk of `automerge-task`: a merge commit is created and pushed, and its associated branch deleted.
+
+TODO:
+A discrepancy occurs between running this task against GitHub and GitHub Enterprise. On GHE, while the commit is merged the PR remains open whereas on GitHub, it is shown as merged. A commented out section in `automerge-task` notes this, which will be investigated further in a separate issue. 
```

Comment:
Does this issue exist - maybe reference it here?

Reply:
https://github.com/rhd-gitops-example/services/issues/76 raised and linked.